### PR TITLE
Fixes + tests exterior derivative

### PR DESCRIPTION
**Pull request by blegouix**
- Fix coboundary to handle exterior sides of the domain.
- Support corner cases (rank = 0 and rank = 1) in AntisymmetricTensorIndex
- Revert `Tensor::access_element<>` consteval
- Generic test for exterior derivative (coalescent or contiguous index, all derivative dimension, all rank until 4D)

### CODE REVIEW

File: src/exterior/coboundary.hpp [hunk 1]
```diff
@@ -136,18 +137,24 @@ coboundary(
...
-                                               ddc::DiscreteElement<CochainTag>(std::distance(
-                                                       lower_chain.begin(),
-                                                       std::
-                                                               find(lower_chain.begin(),
-                                                                    lower_chain.end(),
-                                                                    j->discrete_vector()))));
+                        values[std::distance(simplex_boundary.begin(), j)] = tensor.mem(
+                                misc::domain_contains(tensor.domain(), j->discrete_element())
+                                        ? j->discrete_element()
+                                        : elem, // TODO this is un assumption on boundary condition (free boundary), need to be generalized
```
Comment (blegouix) [inline src/exterior/coboundary.hpp:151]:
```suggestion
                                        : elem, // TODO this is an assumption on boundary condition (free boundary), needs to be generalized
```
